lib/server.js

```diff
@@ -98,6 +100,19 @@ function initSync(mediator, mbaasApi, datasetId, syncOptions) {
   return deferred.promise;
 }
 
+/**
+ * If the objects _id field is a Mongodb~ObjectID, convert it to a string.
+ *
+ * @param {String|Object} object - The object to check.
+ * @returns {String} - The original _id field or the converted Mongodb~ObjectID
+ */
+function getObjectId(object) {
```

Comment:
@nialldonnellyfh @david-martin Just to summarise what we were speaking about. There should be two actions taken here. The first is to document that invalid JSON will not work with the new sync code. The second is to make a change to sync to not allow `_id` fields to be included in a sync record.